backends/gdx-backend-headless/src/com/badlogic/gdx/backends/headless/HeadlessApplicationConfiguration.java

```diff
@@ -16,13 +16,81 @@
 
 package com.badlogic.gdx.backends.headless;
 
+import com.badlogic.gdx.Files;
+
+import java.util.Locale;
+
 public class HeadlessApplicationConfiguration {
 	/** The amount of updates targeted per second. Use 0 to never sleep;  negative to not call the render method at all.
 	 *  Default is 60. */
 	public int updatesPerSecond = 60;
-	/** Preferences directory for headless. Default is ".prefs/". */
-	public String preferencesDirectory = ".prefs/";
+	/** Preferences directory for headless. Default depends on operating system. */
+	public String preferencesDirectory = getDefaultPreferencesDirectory();
+	public Files.FileType preferencesFileType = getDefaultPreferencesFileType();
+	public String preferencesBackupDirectory = ".prefs/";
+	public Files.FileType preferencesBackupFileType = Files.FileType.External;
 
 	/** The maximum number of threads to use for network requests. Default is {@link Integer#MAX_VALUE}. */
 	public int maxNetThreads = Integer.MAX_VALUE;
+
+	private enum OperatingSystem {

```

Comment:
As stated in #6559, I think OS checks should use `UIUtils#isWindows`, etc. instead of duplicating this stuff all over the place.

Reply:
Thanks, will fix.